mm2src/kdf_walletconnect/src/session/rpc/settle.rs

```diff
@@ -51,8 +51,8 @@ pub(crate) async fn reply_session_settle_request(
         session.expiry = settle.expiry;
 
         if let Some(value) = settle.session_properties {
-            let session_properties = serde_json::from_str::<SessionProperties>(&value.to_string())?;
-            session.session_properties = Some(session_properties);
+            let session_properties = serde_json::from_str::<Option<SessionProperties>>(&value.to_string())?;
+            session.session_properties = session_properties;
```

Comment:
actually@mariocynicys 

> do you mean we want to be able to discard this field if we were not able to deserialize it?
i think then the serde_json::from_str::<SessionProperties>(&value.to_string())? should be optional, i.e. we shouldn't fail (?) the whole function if we fail deserializing the arbitrary value.
p.s. we can also use serde_json::from_value instead of from_str and ditch .to_string()

I didn't read this clearly the  last time. 

> do you mean we want to be able to discard this field if we were not able to deserialize it?

this field contains important informations especially for `cosmos` chains. So if  a wallet returns this field in it's session settle response, we should deserialize and return error if something is missing

Reply:
> this field contains important informations especially for cosmos chains. So if a wallet returns this field in it's session settle response, we should deserialize and return error if something is missing

well, if this is the case then my original comment holds https://github.com/KomodoPlatform/komodo-defi-framework/pull/2223#discussion_r1875035917

Reply:
let me resolve this thread and discuss here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2223#discussion_r1875035917 for simplicity.